File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    
    public boolean isAnagram(String s, String t) {
    
        int counts[]=new int[26];
        int countt[]=new int[26];
        int count=0;
        
        if(s.length()!=t.length())
            return false;
        
        for(int i=0;i<s.length();i++)
        {    
            counts[s.charAt(i)-'a']++;
            countt[t.charAt(i)-'a']++;   
        }
        
        for(count=0;count<counts.length;count++)
        {
            if(counts[count]==countt[count])
                continue;
            else
                return false;
        }
        
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        
        var countS=Array(26){0}
        var countT=Array(26){0}
        
        if(t.length!=s.length)
            return false
        
        for(i in 0 until s.length){
            countS[s[i]-'a']++
            countT[t[i]-'a']++
        }
        
        for(i in 0 until 26){
            if(countS[i]!=countT[i])
                return false
        }
        
        return true
    }
}